File: CCC-2000-S1.py
import sys
input = sys.stdin.readline

quarters = int(input())
a, b, c = int(input()), int(input()), int(input())
plays = 0
while quarters >= 1:
        a += 1
        plays += 1
        quarters -= 1
        if a % 35 == 0 and a != 0:
            quarters += 30
        if quarters == 0:
            break

        b += 1
        quarters -= 1
        plays += 1
        if b % 100 == 0 and b != 0:
            quarters += 60
        if quarters == 0:
            break

        quarters -= 1
        c += 1
        plays += 1
        if c % 10 == 0 and c != 0:
            quarters += 9
        if quarters == 0:
            break


print(f"Martha plays {plays} times before going broke.")

File: CCC-2000-S1.cpp
#include <bits/stdc++.h>
using namespace std;
int a, b, c, d;
int main() {
    ios::sync_with_stdio(0); 
    cin.tie(0);
    cin >> a >> b >> c >> d;
    int cnt = 0;
    while (a > 0) {
        a--; b++; cnt++;
        if (b == 35) a += 30; b %= 35;
        if (a <= 0) break;

        a--; c++; cnt++;
        if (c == 100) a += 60; c %= 100;
        if (a <= 0) break;

        a--; d++; cnt++;
        if (d == 10) a += 9; d %= 10;
    }
    cout << "Martha plays " << cnt << " times before going broke.";
    return 0;
}